### PR TITLE
A few Mac-friendly tweaks while I have a chance.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -55,3 +55,4 @@ CMakeLists.txt
 # Let's not even talk about it.
 node_modules
 **/node_modules
+**/package-lock.json
```
Comment (sompylasar) [inline .gitignore]: I see you've upgraded npm >_<

Comment (sompylasar) [reply .gitignore]: By the way, the Node package should have its own gitignore, for modularity. Global gitignore is a "god object".

Comment (dkorolev) [reply .gitignore]: Why not .gitignore all those lockfiles regardless?

Comment (sompylasar) [reply .gitignore]: 1. Not all files with a lock name are lock files in the sense you're thinking.
2. You assume that you "own" all the subdirectories, and the rules equally apply to all of them (which may or may not be true).

Comment (dkorolev) [reply .gitignore]: Explain in one sentence what specific change would you like to see, or greenlight this PR please.

Comment (sompylasar) [reply .gitignore]: If you didn't notice, I have greenlighted.

**Pull request merged**